lib/charms/data_platform_libs/v0/s3.py

```diff
@@ -250,6 +250,41 @@ def _load_relation_data(self, raw_relation_data: dict) -> dict:
                 connection_data[key] = raw_relation_data[key]
         return connection_data
 
+    # def _diff(self, event: RelationChangedEvent) -> Diff:
+    #     """Retrieves the diff of the data in the relation changed databag.
+
+    #     Args:
+    #         event: relation changed event.
+
+    #     Returns:
+    #         a Diff instance containing the added, deleted and changed
+    #             keys from the event relation databag.
+    #     """
+    #     # Retrieve the old data from the data key in the application relation databag.
+    #     old_data = json.loads(event.relation.data[self.local_app].get("data", "{}"))
+    #     # Retrieve the new data from the event relation databag.
+    #     new_data = {
+    #         key: value for key, value in event.relation.data[event.app].items() if key != "data"
+    #     }
+
+    #     # These are the keys that were added to the databag and triggered this event.
+    #     added = new_data.keys() - old_data.keys()
+    #     # These are the keys that were removed from the databag and triggered this event.
+    #     deleted = old_data.keys() - new_data.keys()
+    #     # These are the keys that already existed in the databag,
+    #     # but had their values changed.
+    #     changed = {
+    #         key for key in old_data.keys() & new_data.keys() if old_data[key] != new_data[key]
+    #     }
+
+    #     # TODO: evaluate the possibility of losing the diff if some error
+    #     # happens in the charm before the diff is completely checked (DPE-412).
+    #     # Convert the new_data to a serializable format and save it for a next diff check.
+    #     event.relation.data[self.local_app].update({"data": json.dumps(new_data)})
+
+    #     # Return the diff with all possible changes.
+    #     return Diff(added, changed, deleted)
```

Comment:
Should we just remove this?

Reply:
Sure, go for it.
It is a library coming from [another](https://github.com/canonical/data-platform-libs/blob/main/lib/charms/data_platform_libs/v0/s3.py) repository (removal is not applicable to this PR).

Reply:
Yes, right! Sorry 

The charmlib update are a bit confusing in this respect (what we should review and what we should not). With python dependensies is easier...whatever is in the reqs.txt we don't mind :joy: